### PR TITLE
Update documentation on package resources

### PR DESCRIPTION
**Pull request by Dilhasha**
## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

**Comment by sm1990** [comment]
LGTM

File: swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Writing code in an organized manner from the beginning of the project is importa
 
 Ballerina code is organized in a single shareable unit called a `package`.
```
Comment (MaryamZi) [inline swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md:15]: Not introduced by this PR, but shall we remove backticks around package, module, etc. We use backticks only for keywords, references to names, etc.

File: swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Writing code in an organized manner from the beginning of the project is importa
...
 
-A `package` is a collection of `modules`, and a `module` is a collection of Ballerina source files, test files, and resources. A package should contain at least one module called the default module. Each module has its own directory, which organizes source files, test files, and resources.
+A `package` is a collection of `modules`, and a `module` is a collection of Ballerina source files, and test files. A package should contain at least one module called the default module. Each module has its own directory, which organizes source files, and test files. A package can contain resources and test-only resources specific to the package in the package source root.
```
Comment (MaryamZi) [inline swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md:17]:
```suggestion
A package is a collection of modules, and a `module` is a collection of Ballerina source files and test files. A package should contain at least one module, which becomes the default module. Each module has its own directory, which organizes source files and test files. A package can contain resources and test-only resources specific to the package in the package source root.
```

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md [hunk 3]
```diff
@@ -498,10 +498,14 @@ This file is in markdown format. It will be auto-generated when you create a lib
 
 The `target/` directory contains artifacts generated by building a package.
 
-## The `resources/`, `tests/` directories, and `Module.md`
+## The `tests/` directory, and `Module.md`
```
Comment (MaryamZi) [inline swan-lake/development-tutorials/source-code-dependencies/package-references.md:501]:
```suggestion
## The `tests/` directory and `Module.md` file
```

File: swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md (hunk 1 above)
Comment (Dilhasha) [reply swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md]: Addressed with 3e84be21c2dc33076f5c69ca3fe0b9bba41a2b89

File: swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md [hunk 4]
```diff
@@ -12,9 +12,9 @@ intro: The sections below include information about packages and how you can man
...
 
-Ballerina code is organized in a single shareable unit called a `package`.
+Ballerina code is organized in a single shareable unit called a package.
 
-A `package` is a collection of `modules`, and a `module` is a collection of Ballerina source files, test files, and resources. A package should contain at least one module called the default module. Each module has its own directory, which organizes source files, test files, and resources.
+A package is a collection of modules, and a module is a collection of Ballerina source files and test files. A package should contain at least one module, which becomes the default module. Each module has its own directory, which organizes source files and test files. A package can contain resources and test-only resources specific to the package in the package source root.
```
Comment (MaryamZi) [inline swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md:17]: How is a resource defined as test-only?

File: swan-lake/development-tutorials/source-code-dependencies/package-references.md [hunk 5]
```diff
@@ -546,18 +548,13 @@ The package sources are the `.bal` files in the `root` directory, and `tests/` d
...
-
 ### The `tests/` directory
 
 The `tests/` directory contains unit tests for the module and tests the module in isolation. The module-level test cases have access to the symbols with module-level visibility.
 
-
 ## The `generated/` directory
 
-This directory contains generated Ballerina code. The `.bal` files at the root of the generated directory become a part of the default module. Any direct subdirectory becomes a module in the package. The files will logically merge into the existing modules during compilation.
+This directory contains generated Ballerina code. The `.bal` files at the root of the generated directory become a part of the default module. Any direct subdirectory becomes a module in the package. The files will logically merge into the existing modules during compilation. Any files added to the `resources` folder within the `generated/` directory will be recognized as resources for the package.
```
Comment (MaryamZi) [inline swan-lake/development-tutorials/source-code-dependencies/package-references.md:557]:
```suggestion
This directory contains generated Ballerina code. The `.bal` files at the root of the generated directory become a part of the default module. Any direct subdirectory becomes a module in the package. The files will logically merge into the existing modules during compilation. Any files added to the `resources` directory within the `generated/` directory will be recognized as resources for the package.
```

File: swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md (hunk 4 above)
Comment (Dilhasha) [reply swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md:17]: when the resources are added to a `resources` directory within `tests` directory.

Comment (Dilhasha) [reply swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md:17]:
```suggestion
A package is a collection of modules, and a module is a collection of Ballerina source files and test files. A package should contain at least one module, which becomes the default module. Each module has its own directory, which organizes source files and test files. A package can include resources in a `resources` directory located at the package source root, and test-specific resources in a `resources` directory within the `tests` directory at the package source root.
```

Comment (MaryamZi) [reply swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md]: If we have test-specific resources for submodules, they also have to go in the tests directory in the root?

**Review by MaryamZi — APPROVED** [review]

File: swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md (hunk 4 above)
Comment (Dilhasha) [reply swan-lake/development-tutorials/source-code-dependencies/organize-ballerina-code.md]: we don't support resources for sub modules. Resources are applicable to the entire package not module specific.

**Pull request merged**